iocore/dns/DNS.cc

```diff
@@ -808,6 +808,15 @@ DNSHandler::recv_dns(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
           dnsc->tcp_data.buf_ptr = make_ptr(dnsBufAllocator.alloc());
         }
         if (dnsc->tcp_data.total_length == 0) {
+          // see if TS gets a two-byte size
+          short tmp = 0;
```

Comment:
It should be `uint16_t` because the size is important here.

Reply:
good idea! will fix.